tests/test_functions.py

```diff
@@ -0,0 +1,119 @@
+import requests, difflib, sys
+import re
+
+command_line_arguments = []
+
+number_of_errors = 0
```

Comment:
Again this needn't be a global variable, functions should return errors or raise exceptions when they fail and those can be caught by the test

Reply:
I created a global variable so that it can simply print the error, and then it raises an exception if any of them have failed, with the number of errors it got.
Is there a better way to do this?

Reply:
I guess what I'm saying is that I think these functions should probably not hold any sort of state between executions. 

You could have code in your `firsttimesetup` module that tracks whether these raise an exception or not. These functions could pass exceptions up to that module. Then you could have a command-line option that specifies whether or the test suite should keep-going in an error case.

Reply:
Yes, I agree that it's generally bad practice.
Hmm, not sure what you mean. If it passes the exception back up, then whoever is writing tests has to keep track of exceptions in the response and add to their counter, which I don't like so much, since I am trying to make it as simple as possible to write tests. Is there a way around this?